test/integration/etcd/data.go

```diff
@@ -742,6 +755,33 @@ func validateStorageData(etcdStorageData map[schema.GroupVersionResource]Storage
 	}
 }
 
+// storageVersionAtEmulationVersion tries to find the correct storage version at an emulation version.
+// If a GVK is introduced after the min compatibility version, we need to use an earlier version in storage.
+func storageVersionAtEmulationVersion(key schema.GroupVersionResource, expectedGVK *schema.GroupVersionKind, emuVer string, etcdStorageData map[schema.GroupVersionResource]StorageData) string {
+	// expectedGVK is needed to find the correct GVK with the correct storage version.
+	if expectedGVK == nil {
+		return ""
+	}
+	minCompatVer := version.MustParse(emuVer).SubtractMinor(1)
```

Comment:
@siyuanfoundation heads up that this might need adjusting for alternate min-compat-versions in the future (but that's fine)